src/components/_dialog.scss

```
@@ -0,0 +1,212 @@
+      しかし、民家で史料が発見されるなど研究も進み、少なくとも江戸時代には確実に存在したとされている。<br>
+      なお、「甲賀流」という名称の単一の流派は存在せず、あくまで甲賀に伝わる複数の流派があわさって甲賀流と呼ばれているにすぎない。<br>
+    </div>
+  </div>
+  <div class="ncgr-dialog__footer">
+    <button class="ncgr-button ncgr-button-default ncgr-button--fluid ncgr-button--light">とじる</button>
+  </div>
+</div>
+```
+
+*/
+
+.ncgr-dialog {
+  width: 40rem;
+  height: auto;
+  max-height: calc(100vh - 12rem);
+  position: fixed;
+  top: 50vh;
+  left: calc(50vw - 20rem);
+  transform: translateY(-50%);
+  background-color: var(--color-secondary-grouped-background);
+  border-radius: 5px;
+  overflow: hidden;
+  display: none;
+  flex-direction: column;
+  z-index: 2147483645;
+  & * {
+    z-index: 2147483645;
+  }
+  @include mediaquery(mobile) {
+    height: auto;
+    width: calc(100vw - 3rem);
+    left: 1.5rem;
+  }
+  &.-active {
+    display: flex;
+    animation: dialog-fade-in 0.5s cubic-bezier(0.4, 0, 0.2, 1),
+    dialog-fade-in-shadow 0.5s ease;
+    @include shadow-16dp();
+    @include mediaquery(mobile) {
+      animation: dialog-fade-in-mobile 0.5s cubic-bezier(0.4, 0, 0.2, 1),
+      dialog-fade-in-shadow 0.5s ease;
+    }
+    * {
+      animation: dialog-fade-in-sheet-inner 0.5s ease;
+    }
+  }
+}
+
+.ncgr-dialog__close {
+  position: absolute;
+  top: 0.5rem;
+  left: 0.5rem;
+  width: 2.5rem;
+  height: 2.5rem;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  background:rgba(0, 0, 0, 0.2);
+  border-radius: 50%;
+  margin: 0;
+  font-size: 2rem;
+  color: var(--color-white);
+  z-index: 2147483646;
+}
+
+.ncgr-dialog__body {
+  display: inline-block;
+  position: relative;
+  width: 40rem;
+  height: auto;
+  flex-grow: 1;
+  overflow-x: hidden;
+  overflow-y: scroll;
+  @include mediaquery(tablet) {
+    width: 100%;
+  }
+}
+
+.ncgr-dialog__footer {
+  width: 100%;
+  height: 4rem;
+  display: flex;
+  justify-content: center;
+  align-items: center;
+  box-sizing: border-box;
+  padding: 0.5rem;
+}
+
+// animation
+
+@keyframes dialog-fade-in {
+
+  0% {
+    width: 0;
+    height: 10px;
+    left: 50%;
+  }
+
+  40% {
+    width: 40rem;
+    height: 10px;
+    left: calc(50vw - 20rem);
+    @include mediaquery(mobile) {
+      width: calc(100vw - 3rem);
+      height: 10px;
+      left: 1.5rem;
+    } 
+  }
+
+  100% {
+    width: 40rem;
+    height: auto;
+    left: calc(50vw - 20rem);
+    @include mediaquery(mobile) {
+      width: calc(100vw - 3rem);
+      height: auto;
+      left: 1.5rem;
+    } 
+  }
+}
+
+@keyframes dialog-fade-in-mobile {
+
+  0% {
+    width: 0;
+    height: 10px;
+    left: 50%;
+  }
+
+  40% {
+    width: calc(100vw - 3rem);
+    height: 10px;
+    left: 1.5rem;
+  }
+
+  100% {
+    width: calc(100vw - 3rem);
+    height: auto;
+    left: 1.5rem;
+  }
+}
+
+@keyframes dialog-fade-in-inner {
+
+  0% {
+    opacity: 0;
+  }
+
+  40% {
+    opacity: 0;
+  }
+
+  100% {
+    opacity: 1;
+  }
+}
+
+@keyframes dialog-fade-in-shadow {
```

[IMO] こっちのほうが上の文脈からすると自然かも

```suggestion
@keyframes dialog-shadow-fade-in {
```